File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> prodTillIdx(n, 1), prodAfterIdx(n, 1), res(n,0);

        for(int i = 1; i < n; i++){
            prodTillIdx[i] = prodTillIdx[i-1] * nums[i-1];
        }

        for(int i = n-2; i >= 0; i--){
            prodAfterIdx[i] = prodAfterIdx[i+1] * nums[i+1];
        }

        for(int i = 0; i < n; i++){
            res[i] = prodTillIdx[i] * prodAfterIdx[i];
        }

        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqArr = {}
        for i in range(len(nums)):
            if nums[i] not in freqArr.keys():
                freqArr[nums[i]] = 1
            else:
                freqArr[nums[i]] += 1
        freqArr = dict(sorted(freqArr.items(), key = lambda item: item[1], reverse=True))
        
        res = []
        for i in range(k):
            '''print((freqArr.keys())[i])'''
            res.append(list(freqArr.keys())[i])
        return res